File: benchmarks/eval_headtohead/corpus/go/worker.go
// Background worker — a real caller of ProcessOrder (D2 ground truth).
package order

func Drain(queue []map[string]string) []map[string]string {
	out := make([]map[string]string, 0, len(queue))
	for _, order := range queue {
		out = append(out, ProcessOrder(order))
	}
	return out
}

File: benchmarks/eval_headtohead/corpus/rust/worker.rs
//! Background worker — a real caller of process_order (D2 ground truth).

use crate::core::process_order;
use std::collections::HashMap;

pub fn drain(queue: Vec<HashMap<String, String>>) -> Vec<HashMap<String, String>> {
    queue.iter().map(process_order).collect()
}
